### DISCUSSION TITLE
Touched behaviour of checkboxes

### DISCUSSION THREAD

**Discussion by maurer2** [react-hook-form/react-hook-form#12064] [General]
Hello, I noticed that when toggling a checkbox in a browser on MacOS the "touched" flag doesn't seem to change. This issue was previously reported for Safari here https://github.com/react-hook-form/react-hook-form/issues/2084, but it seems to be happening for others browsers on MacOS as well now. According to this issue https://github.com/whatwg/html/issues/4356 the blur event used to work correctly in Firefox on MacOs for checkboxes, but that seems to have changed.
I wonder if there is a good way to still set the touched flag when toggling a checkbox?

Thanks

Test repo: https://github.com/maurer2/rhf-output-schema

## Simplified form markup

```tsx
export default function FormFields() {
  const {
    register,
    handleSubmit,
    formState: { errors },
    control,
  } = useForm{
    defaultValues: {
      name: "Name1 Name2",
      age: "50",
      isActive: true,
    },
  });

  const onSubmit  = (data) => {
    console.log(data);
  };

  return (
    <>
      <form
        className="form"
        name="test form"
        aria-label="test form"
        onSubmit={handleSubmit(onSubmit)}
      >
        <div className="field">
          <label htmlFor="name">Name</label>
          <input {...register("name")} id="name" />
          <ErrorMessage errors={errors} name="name" />
        </div>

        <div className="field">
          <label htmlFor="age">Age</label>
          <input {...register("age")} id="age" />
          <ErrorMessage errors={errors} name="age" />
        </div>

        <div className="field-checkbox">
          <input type="checkbox" {...register("isActive")} id="isActive" />
          <label htmlFor="isActive">is active</label>
          <div className="error">
            <ErrorMessage errors={errors} name="isActive" />
          </div>
        </div>

        <button type="submit">Send</button>
      </form>
      <DevTool control={control} />
    </>
  );
} 
```

### Default state

<img width="990" alt="before" src="https://github.com/react-hook-form/react-hook-form/assets/6861911/71795ec1-254f-424e-a9e5-3b1f147b9f1a">

### After toggling the checkbox
<img width="990" alt="after" src="https://github.com/react-hook-form/react-hook-form/assets/6861911/39a1d866-2af0-48d6-8fb0-8b5466fc2d79">

### After toggling the checkbox again
<img width="993" alt="image" src="https://github.com/react-hook-form/react-hook-form/assets/6861911/22c86b50-6ef8-4065-93a2-acb729e5917f">

**Comment by daxaxelrod** [comment]
Here I was thinking i was crazy. Facing the same thing
